src/popup/components/TransactionList.vue

```diff
@@ -50,179 +40,230 @@ import {
+        });
       }
+      return transactionList;
     }
 
-    async function loadMore() {
-      if (loading.value) return;
+    function filterTransactionsByDisplayMode(transactionList: ICommonTransaction[]) {
+      return transactionList.filter((transaction) => {
+        const outerTx = transaction.tx!;
+        const innerTx = transaction.tx ? getInnerTransaction(transaction.tx) : null;
+
+        const txOwnerAddress = getTxOwnerAddress(innerTx);
+
+        const direction = getTxDirection.value(
+          outerTx.payerId ? outerTx : innerTx,
+          (transaction as ITransaction).transactionOwner
+          || (
+            getOwnershipStatus(activeAccount.value, accounts.value, innerTx)
+            !== TRANSACTION_OWNERSHIP_STATUS.current
+            && txOwnerAddress
+          ),
+        );
+
+        const isDex = isTxDex(innerTx, getDexContracts.value);
+
+        switch (displayMode.value.key) {
+          case FILTER_MODE.all:
+            return true;
+          case FILTER_MODE.dex:
+            return isDex;
+          case FILTER_MODE.out:
+            return direction === TX_DIRECTION.sent && !isDex;
+          case FILTER_MODE.in:
+            return direction === TX_DIRECTION.received;
+          default:
+            throw new Error(`${i18n.t('pages.recentTransactions.unknownMode')} ${displayMode.value.key}`);
+        }
+      });
+    }
+
+    function filterTransactionsBySearchPhrase(transactionList: ICommonTransaction[]) {
+      return transactionList.filter(
+        (transaction) => !searchPhrase.value || getTxSymbol.value(transaction)
+          .toLocaleLowerCase()
+          .includes(searchPhrase.value.toLocaleLowerCase()),
+      );
+    }
+
+    function sortTransactionListByDate(transactionList: ICommonTransaction[]) {
+      return transactionList.sort(sortTransactionsByDateCallback);
+    }
+
+    const filteredTransactions = computed(
+      () => pipe<ICommonTransaction[]>([
+        narrowTransactionsToDefinedToken,
+        filterTransactionsByDisplayMode,
+        filterTransactionsBySearchPhrase,
+        sortTransactionListByDate,
+      ])(loadedTransactionList.value),
+    );
+
+    async function fetchTransactionList(recent?: boolean) {
       loading.value = true;
       try {
-        await root.$store.dispatch('fetchTransactions', { limit: TXS_PER_PAGE, address: currentAddress.value });
+        await fetchTransactions({
+          limit: TXS_PER_PAGE,
+          recent,
+          address: currentAddress.value,
+        });
       } finally {
         loading.value = false;
       }
-      checkLoadMore();
     }
 
-    async function getLatest() {
-      try {
-        await root.$store.dispatch('fetchTransactions', { limit: 10, recent: true, address: currentAddress.value });
-      } finally {
-        loading.value = false;
+    async function loadMore() {
+      if (loading.value) {
+        return;
+      }
+      await fetchTransactionList();
```

Comment:
I don't see a point in this return 
```suggestion
      if (!loading.value) {
        await fetchTransactionList();
      }
```